pdns/recursordist/docs/lua-scripting/ffi.rst

```diff
@@ -0,0 +1,143 @@
+
+   Fill out `addr` with sender's IP address
+
+.. function:: pdns_ffi_param_get_qname_raw(pdns_ffi_param_t* ref, const char** qname, size_t* qnameSize) -> void
+
+   Fill out `qname` with query's qualified name
+
+.. function:: pdns_ffi_param_get_local_raw(pdns_ffi_param_t* ref, const void** addr, size_t* addrSize) -> void
+
+   Fill out `addr` with local IP address the query was received on
+
+.. function:: pdns_ffi_param_set_tag(pdns_ffi_param_t* ref, unsigned int tag) -> void
+
+   Tag the query with the given number
+
+.. function:: pdns_ffi_param_add_policytag(pdns_ffi_param_t* ref, const char* name) -> void
+
+   Add the given tag to the query
+
+.. function:: pdns_ffi_param_set_requestorid(pdns_ffi_param_t* ref, const char* name) -> void
+
+   Set query's requestor ID
+
+.. function:: pdns_ffi_param_set_devicename(pdns_ffi_param_t* ref, const char* name) -> void
+
+   Set query's device name
+
+.. function:: pdns_ffi_param_set_deviceid(pdns_ffi_param_t* ref, size_t len, const void* name) -> void
+
+   Set query's device ID
+
+.. function:: pdns_ffi_param_set_routingtag(pdns_ffi_param_t* ref, const char* name) -> void
+
+   Set routing tag which is used as an extra name to identify records in the record cache, see :func:`gettag`
+
+.. function:: pdns_ffi_param_set_variable(pdns_ffi_param_t* ref, bool variable) -> void
+
+   Mark as variable and ensure it’s not inserted into the packetcache
+
+.. function:: pdns_ffi_param_set_ttl_cap(pdns_ffi_param_t* ref, uint32_t ttl) -> void
+
+   Cap the max TTL of the returned records
+
+.. function:: pdns_ffi_param_set_log_query(pdns_ffi_param_t* ref, bool logQuery) -> void
+
+   Turn on/off query logging
+
+.. function:: pdns_ffi_param_set_log_response(pdns_ffi_param_t* ref, bool logResponse) -> void
+
+   Turn on/off response logging
+
+.. function:: pdns_ffi_param_set_rcode(pdns_ffi_param_t* ref, int rcode) -> void
+
+   Set response RCode
+
+.. function:: pdns_ffi_param_set_follow_cname_records(pdns_ffi_param_t* ref, bool follow) -> void
+
+   Instruct the recursor to do a proper resolution in order to follow any CNAME records added
+
+.. function:: pdns_ffi_param_set_extended_error_code(pdns_ffi_param_t* ref, uint16_t code) -> void
+
+   Set extended DNS error info code
+
+.. function:: pdns_ffi_param_set_extended_error_extra(pdns_ffi_param_t* ref, size_t len, const char* extra) -> void
+
+   Set extended DNS error extra text
+
+.. function:: pdns_ffi_param_set_padding_disabled(pdns_ffi_param_t* ref, bool disabled) -> void
+
+   Disable padding
+
+.. function:: pdns_ffi_param_add_record(pdns_ffi_param_t* ref, const char* name, uint16_t type, uint32_t ttl, const char* content, size_t contentSize, pdns_record_place_t place) -> bool
+
+   Adds a record. Returns true if it was correctly added, false otherwise
+
+.. function:: pdns_ffi_param_add_meta_single_string_kv(pdns_ffi_param_t* ref, const char* key, const char* val) -> void
+
+    .. versionadded:: 4.5.3
```

Comment:
Does this need a backport? If so, the earliest version it's going to be in is 4.5.5, as 4.5.4 was just released without this. Otherweise the `versionadded::`  should be 4.6.0